.github/workflows/validate-and-execute-notebooks.yml

```diff
@@ -0,0 +1,122 @@
+name: Smoke Tests for Notebooks
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - "notebooks/**/*.ipynb"
+      - ".github/workflows/validate-and-execute-notebooks.yml"
+  push:
+    branches: [ main ]
+    paths:
+      - "notebooks/**/*.ipynb"
+      - ".github/workflows/validate-and-execute-notebooks.yml"
+
+permissions:
+  contents: read
+
+jobs:
+  validate_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Set the notebooks to validate, wildcards are allowed
+        notebooks_to_validate: ["notebooks/**/*.ipynb"]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: pip
+
+      - name: Install Testing Tools
+        run: |
+          pip install nbformat ipykernel
+          ipython kernel install --name "python3" --user
+
+      - name: Validate Notebooks
+        run: |
+          set -ux
+
+          shopt -s globstar nullglob
+
+          for FILE in ${{ matrix.notebooks_to_validate }}; do
+            if [ -f "$FILE" ]; then
+              echo "Validating notebook '$FILE'..."
+
+              FILE="$FILE" python - <<'PY'
```

Comment:
I'm not a fan of having a python script in line within the action. As follow up work could we split this out into it's own file. I really like the `nbformat.validate()` check here.